src/dataflow-types/src/client/replicated.rs

```diff
@@ -194,7 +260,19 @@ where
                 let mut stream: tokio_stream::StreamMap<_, _> = self
                     .replicas
                     .iter_mut()
-                    .map(|(id, shard)| (id.clone(), shard.as_stream()))
+                    .map(|(id, (_, rx))| {
+                        let rx_stream = Box::pin(async_stream::stream! {
+                            loop {
+                                match rx.recv().await {
+                                    Some(Ok(response)) => yield Ok(response),
+                                    Some(Err(error)) => yield Err(error),
+                                    None => { return; }
+                                }
+                            }
+                        });
```

Comment:
If you make `rx` and `UnboundedReceiverStream` as mentioned above, I think you can ditch all this.